### DISCUSSION TITLE
deepmd pair-style in LAMMPS

### DISCUSSION THREAD

**Discussion by omidshy** [deepmodeling/deepmd-kit#2724] [Ideas]
If a set of models are provided to `pair_style deepmd ...` command, only the first model is used for energy and force prediction at each MD step, and the following models are used to compute the model deviation.

Having the option to use the average of the predictions from a set of models to run MD would be a practical feature, since the ensemble averages can provide a more robust prediction, specially in parts of the configuration space that training data are scarce.

**Comment by wanghan-iapcm** [comment]
By the efficiency consideration, the model deviation is calculated every `out_freq` steps, thus at`out_freq-1` steps only one model is evaluate, and at one step the ensemble of model is evaluated. The average of model ensemble may be biased from the dft, especially in the configuration space not well covered by the training data, thus using the average may not necessarily lead to a better prediction.

**Comment by njzjz** [comment]
I've added a model type in #2781 that can linearly combine a set of models.
